activesupport/lib/active_support/messages/metadata.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+require "time"
+
+module ActiveSupport
+  module Messages
+    class Metadata #:nodoc:
+      def initialize(expires_at, purpose)
+        @expires_at, @purpose = expires_at, purpose
+      end
+
+      class << self
+        def wrap(message, expires_at: nil, expires_in: nil, purpose: nil)
+          { "value" => message, "_rails" => { "exp" => pick_expiry(expires_at, expires_in), "pur" => purpose } }
+        end
+
+        def verify(message, purpose)
+          metadata = extract_metadata(message)
+          metadata.nil? ? message : (metadata.match?(purpose) && metadata.fresh?) ? message["value"] : nil
```

Comment:
We prefer standard ifs to ternaries (especially the hard to read double-ternaries). This would be much clearer to read (albeit the extra lines):

```ruby
metadata = extract_metadata(message)

if metadata && metadata.match?(purpose) && metadata.fresh?
  message["value"]
else
  message
end
```